mysticeti-core/src/consensus/linearizer.rs

```diff
@@ -41,6 +42,28 @@ impl CommittedSubDag {
         }
     }
 
+    pub fn new_from_commit_data(commit_data: CommitData, block_store: &BlockStore) -> Self {
+        let mut leader_block_idx = None;
+        let blocks = commit_data
+            .sub_dag
+            .into_iter()
+            .enumerate()
+            .map(|(idx, block_ref)| {
+                let block = block_store
+                    .get_block(block_ref)
+                    .expect("We should have the block referenced in the commit data");
+                if block_ref == commit_data.leader {
+                    leader_block_idx = Some(idx);
+                }
+                block
+            })
+            .collect::<Vec<_>>();
+        let leader_block_idx = leader_block_idx.expect("Leader block must be in the sub-dag");
+        let leader_block_ref = blocks[leader_block_idx].reference();
+        let timestamp_ms = (blocks[leader_block_idx].meta_creation_time_ns() / 1000) as u64;
```

Comment:
We should add TODO here to to replace this logic when we work on proper commit timestamps (and ideally cross-reference this in other place where we convert meta_creation_time_ns into timestamp, looks like there are two places now

Reply:
Added meta_creation_time_ms() API